server/src/weaverbird/backends/pandas_executor/steps/utils/condition.py

```diff
@@ -16,6 +19,18 @@
 from weaverbird.pipeline.dates import RelativeDate
 
 
+def _date_bound_condition_to_tz_aware_timestamp(value: str | RelativeDate | datetime) -> Timestamp:
+    if isinstance(value, RelativeDate):
+        value = evaluate_relative_date(value)
+    if isinstance(value, datetime):
+        tz = value.tzinfo or "UTC"
+        # Cannot pass a tz-aware datetime object with tz arg
+        return Timestamp(value.replace(tzinfo=None), tz=tz)
+    else:  # str
+        ts = Timestamp(value)
+        return ts if ts.tzinfo else ts.replace(tz="UTC")
```

Comment:
```suggestion
    # str
    ts = Timestamp(value)
    return ts if ts.tzinfo else ts.replace(tz="UTC")
```

Reply:
Linters prefer this, but our style in the various repos is to have an `else` branch even when returning in previous branches